src/Tokenizer/TokensAnalyzer.php

```diff
@@ -205,60 +206,56 @@ public function isBlockMultiline(Tokens $tokens, int $index): bool
 
     /**
      * @param int $index Index of the T_FUNCTION token
-     *
-     * @return array{visibility: null|T_PRIVATE|T_PROTECTED|T_PUBLIC, static: bool, abstract: bool, final: bool}
      */
-    public function getMethodAttributes(int $index): array
+    public function getMethodAnalysis(int $index): MethodAnalysis
```

Comment:
I know this class is marked as `@internal` but shouldn't we provide fallback method anyway? I believe this can break people's custom fixers.

Reply:
Isn't the whole purpose of the `@internal` tag to not care about such changes? If we keep such fallback methods for everything we would pollute the codebase with a lot of garbage.

Reply:
Personally I think Fixer should provide public API and tools for people who create custom fixers. Yeah, we can mark everything as internal and not care, but if we change the API frequently, maintainers of customs fixers and people using them will get frustrated, rightfully 😉. I don't know what the approach was before, but I have a gut feeling that we should keep BC promise in analysers... @keradus what do you think?

Reply:
we didn't had a good discussion about how to provide analysers to devs as library, and we know we have challenges around them - eg have results of them cached.

with that, as we were not happy with state of analysers, we marked it as internal, so we can have our logic here, without messing up with `Tokens`.

when we will have proper approach for analysers, we can drop internal flag.

as it's internal, we are OK to change interface.